juju/testing/apiserver.go

```diff
@@ -359,7 +359,7 @@ func (s *ApiServerSuite) setupApiServer(c *gc.C, controllerCfg controller.Config
 	c.Assert(err, jc.ErrorIsNil)
 	agentAuthFactory := authentication.NewAgentAuthenticatorFactory(systemState, nil)
 
-	authenticator, err := stateauthenticator.NewAuthenticator(context.Background(), cfg.StatePool, systemState, factory.ControllerConfig(), factory.Access(), factory.Macaroon(), agentAuthFactory, cfg.Clock)
+	authenticator, err := stateauthenticator.NewAuthenticator(context.Background(), cfg.StatePool, systemState, systemState.ModelUUID(), factory.ControllerConfig(), factory.Access(), factory.Macaroon(), agentAuthFactory, cfg.Clock)
```

Comment:
This is available in `cfg`, so you might as well get it from there now and remove an access to legacy state.

Reply:
Nice one